fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogBottomSheet.kt

```diff
@@ -116,121 +100,297 @@ fun TranslationsDialogBottomSheet(
         )
 
         TranslationsDialogHeader(
-            showFirstTimeTranslation = showFirstTimeTranslation,
+            title = if (
+                translationsDialogState.isTranslated && translationsDialogState.translatedPageTitle != null
+            ) {
+                translationsDialogState.translatedPageTitle
+            } else {
+                getTranslationsDialogTitle(
+                    showFirstTime = showFirstTime,
+                )
+            },
             onSettingClicked = onSettingClicked,
         )
 
         Spacer(modifier = Modifier.height(8.dp))
 
-        if (showFirstTimeTranslation) {
+        if (showFirstTime) {
             TranslationsDialogInfoMessage(
                 learnMoreUrl = learnMoreUrl,
                 onLearnMoreClicked = onLearnMoreClicked,
             )
         }
 
-        translationError?.let {
-            TranslationErrorWarning(translationError)
-        }
+        DialogContentBaseOnTranslationState(
+            translationsDialogState = translationsDialogState,
+            learnMoreUrl = learnMoreUrl,
+            onLearnMoreClicked = onLearnMoreClicked,
+            onPositiveButtonClicked = onPositiveButtonClicked,
+            onNegativeButtonClicked = onNegativeButtonClicked,
+            onFromSelected = onFromSelected,
+            onToSelected = onToSelected,
+        )
+    }
+}
 
+/**
+ * Dialog content will adapt based on the [TranslationsDialogState].
+ *
+ * @param translationsDialogState The current state of the Translations bottom sheet dialog.
+ * @param learnMoreUrl The learn more link for translations website.
+ * @param onLearnMoreClicked Invoked when the user clicks on the learn more button.
+ * @param onPositiveButtonClicked Invoked when the user clicks on the positive button.
+ * @param onNegativeButtonClicked Invoked when the user clicks on the negative button.
+ * @param onFromSelected Invoked when the user selects an item on the from dropdown.
```

Comment:
```suggestion
 * @param onFromDropdownSelected Invoked when the user selects an item on the from dropdown.
```
Similar to `onNegativeButtonClicked` naming, we should be clear about what the UI element is.

Reply:
Done